playground/backend/internal/setup_tools/builder/setup_builder.go

```diff
@@ -31,73 +31,134 @@ const (
 	javaLogConfigFilePlaceholder = "{logConfigFile}"
 )
 
-// SetupExecutorBuilder return executor with set args for validator, preparator, compiler and runner
-func SetupExecutorBuilder(lc *fs_tool.LifeCycle, pipelineOptions string, sdkEnv *environment.BeamEnvs) (*executors.ExecutorBuilder, error) {
+// SetupValidatorBuilder return executor with set args for validator
+func SetupValidatorBuilder(lc *fs_tool.LifeCycle, sdkEnv *environment.BeamEnvs) (*executors.ExecutorBuilder, error) {
 	sdk := sdkEnv.ApacheBeamSdk
-
-	if sdk == pb.Sdk_SDK_JAVA {
-		pipelineOptions = utils.ReplaceSpacesWithEquals(pipelineOptions)
-	}
-
 	val, err := utils.GetValidators(sdk, lc.GetAbsoluteSourceFilePath())
 	if err != nil {
 		return nil, err
 	}
+	builder := executors.NewExecutorBuilder().
+		WithValidator().
+		WithSdkValidators(val).
+		ExecutorBuilder
+	return &builder, err
+}
+
+// SetupPreparatorBuilder return executor with set args for preparator
+func SetupPreparatorBuilder(lc *fs_tool.LifeCycle, sdkEnv *environment.BeamEnvs) (*executors.ExecutorBuilder, error) {
+	sdk := sdkEnv.ApacheBeamSdk
 	prep, err := utils.GetPreparators(sdk, lc.GetAbsoluteSourceFilePath())
 	if err != nil {
 		return nil, err
 	}
-	executorConfig := sdkEnv.ExecutorConfig
 	builder := executors.NewExecutorBuilder().
-		WithExecutableFileName(lc.GetAbsoluteExecutableFilePath()).
-		WithWorkingDir(lc.GetAbsoluteBaseFolderPath()).
-		WithValidator().
-		WithSdkValidators(val).
 		WithPreparator().
 		WithSdkPreparators(prep).
+		ExecutorBuilder
+	return &builder, err
+}
+
+// SetupCompilerBuilder return executor with set args for compiler
+func SetupCompilerBuilder(lc *fs_tool.LifeCycle, sdkEnv *environment.BeamEnvs) *executors.ExecutorBuilder {
+	sdk := sdkEnv.ApacheBeamSdk
+	executorConfig := sdkEnv.ExecutorConfig
+	builder := executors.NewExecutorBuilder().
 		WithCompiler().
 		WithCommand(executorConfig.CompileCmd).
+		WithWorkingDir(lc.GetAbsoluteBaseFolderPath()).
 		WithArgs(executorConfig.CompileArgs).
 		WithFileName(lc.GetAbsoluteSourceFilePath()).
+		ExecutorBuilder
+
+	switch sdk {
+	case pb.Sdk_SDK_JAVA:
+		builder.
+			WithCompiler().
+			WithFileName(GetFirstFileFromFolder(lc.GetAbsoluteSourceFolderPath()))
+	}
+	return &builder
+}
+
+// SetupRunBuilder return executor with set args for runner
+func SetupRunBuilder(lc *fs_tool.LifeCycle, pipelineOptions string, sdkEnv *environment.BeamEnvs) (*executors.ExecutorBuilder, error) {
+	sdk := sdkEnv.ApacheBeamSdk
+
+	if sdk == pb.Sdk_SDK_JAVA {
+		pipelineOptions = utils.ReplaceSpacesWithEquals(pipelineOptions)
+	}
+	executorConfig := sdkEnv.ExecutorConfig
+	builder := executors.NewExecutorBuilder().
 		WithRunner().
+		WithExecutableFileName(lc.GetAbsoluteExecutableFilePath()).
```

Comment:
Shouldn't we move this to the `pb.Sdk_SDK_PYTHON` case to keep consistency?

Reply:
Done